vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/impl/ApolloWSConnectionHandler.java

```diff
@@ -94,7 +97,7 @@ private void handleMessage(JsonObject jsonObject) {
       return;
     }
 
-    ApolloWSMessage message = new ApolloWSMessageImpl(serverWebSocket, type, jsonObject);
+    ApolloWSMessage message = new ApolloWSMessageImpl(serverWebSocket, type, jsonObject, null);
```

Comment:
Change the `message` type to `ApolloWSMessageImpl` and it won't be necessary to expose the future anymore.

Reply:
Can you add another constructor which does not take a `connectionParams` parameter and initializes the field to `null`?